### ISSUE TITLE
PyTorch code version high FER in completely unsupervised settings?

### ISSUE THREAD

**Issue by JeromeNi** [gary083/GAN_Harmonized_with_HMMs#4]
@SungFeng-Huang I have been playing with your pytorch code version for the first iteration of GAN training using oracle bounds or the provided "gas" bounds. It seems that I wasn't able to achieve the FER numbers reported in the paper, even under the "matching" setting between audio and text. While the oracle bounds gave somewhat more reasonable results, the FER was still in the low 30s%, with a final PER of 30.5%. With the provided unsupervised "gas" bounds, the FER was much much higher than the results in the paper. The FER I could obtain was 70-85%, and the best PER I could get was 65%.

I compared the gas bounds with the oracle bounds and the r-value seemed reasonable (81.77) under a 2-frame (standard 25ms/10ms shift) tolerance window.

I noticed that in the commit comments you also mentioned this issue. Have you ever figured out the reason?

Thanks for the help!

**Comment by SungFeng-Huang** [comment]
Hi @JeromeNi, I'm not sure what's the reason for your problem, but I could share my experience with the comment you noticed.
At that time, my mask function masked out the wrong dimension, which leads to the poor FER at around 60-70%.
I hope my experience helps!
If you have more implementation problems, I think @darongliu might help you more since he is now working on a related project.

**Comment by JeromeNi** [comment]
Hi, could you elaborate more on your issue with the mask function? Do you still remember which part of the code it was from? 

I don't think I have modified any part of the latest code from pre-processing to training. The strange thing was that it only happened when using unsupervised GAS bounds, not to oracle bounds. 

Thanks!

**Comment by SungFeng-Huang** [comment]
The problem I encountered is that I used a mask function for tensors with shape (N,C,L), while I have the input feature with shape (N,L,C) in my code. After fixing this, I got reasonable results, so I'm not sure what happened to your code for getting worse results with GAS bounds.
